pubtools/pulplib/_impl/model/unit/file.py

```diff
@@ -31,6 +32,17 @@ class FileUnit(Unit):
         default="iso", type=str, pulp_field="_content_type_id"
     )
 
+    repository_memberships = pulp_attrib(
+        default=attr.Factory(FrozenList),
```

Comment:
Question: this repository_memberships is only filled in for certain kinds of searches, right?  You don't get this info when you search for units via Repository.search_content?

I think then the default should rather be `None`.  If it's an empty list then there is no obvious way for a caller to tell if they got an empty repository_memberships because the unit genuinely belongs to no repository or because of how the unit was retrieved.

So, the default would be None and the docs would point out that the info is not always available.

```suggestion
        default=None,
```

Reply:
yeah, this makes sense.